scripts/commands/addcurrency.lua

```diff
@@ -9,33 +9,125 @@ cmdprops =
     parameters = "sis"
 };
 
-function onTrigger(player,currency,amount,target)
-    local targ = nil
+function error(player, msg)
+    player:PrintToPlayer(msg);
+    player:PrintToPlayer("@addcurrency <currency type> <amount> {player}");
+end;
 
-    if (currency == nil or amount == nil) then
-        player:PrintToPlayer( "You must enter a valid currency and amount." );
-        player:PrintToPlayer( "@addcurrency <currency type> <amount> <target player>" );
-        return
-    end
+function onTrigger(player,currency,amount,target)
+    local currencies = {
+        sandoria_cp=true,
```

Comment:
If you put in, say, "FOO" as the currency name, then targ:getCurrency("FOO") returns 0, rather than nil.  It's indistinguishable from a valid currency the player's out of, except that it sends a SQL error to the console.  Then the code happily continues, targ:addCurrency("FOO",amount), which sends another error to the console.  Server doesn't crash, but checking for valid currency name lua-side prevents the two console error messages.